paper/src/main/java/com/denizenscript/denizen/paper/utilities/PaperAPIToolsImpl.java

```diff
@@ -109,13 +110,23 @@ public void setSignLine(Sign sign, int line, String text) {
     }
 
     @Override
-    public void sendResourcePack(Player player, String url, String hash, boolean forced, String prompt) {
-        if (prompt == null && !forced) {
-            super.sendResourcePack(player, url, hash, false, null);
+    public void setResourcePack(Player player, String url, String hash, boolean forced, String prompt, String id) {
+        if (prompt == null && !forced && id == null) {
+            super.setResourcePack(player, url, hash, false, null, null);
         }
-        else {
+        else if (id == null) {
             player.setResourcePack(url, CoreUtilities.toLowerCase(hash), forced, PaperModule.parseFormattedText(prompt, ChatColor.WHITE));
         }
+        else {
+            UUID packUUID;
+            try {
+                packUUID = UUID.fromString(id);
+            }
+            catch (IllegalArgumentException e) {
+                packUUID = UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8));
+            }

```

Comment:
I'd probably put this in a util method on `ResourcePackCommand`, just because it's repeated a couple times